sktime/classification/kernel_based/_rocket_classifier.py

```diff
@@ -49,13 +47,12 @@ class ROCKETClassifier(BaseClassifier):
     tsml/classifiers/shapelet_based/ROCKETClassifier.java
     """
 
-    # Capability tags
-    capabilities = {
-        "multivariate": True,
-        "unequal_length": False,
-        "missing_values": False,
-        "train_estimate": False,
-        "contractable": False,
```

Comment:
Hello, Matthew,

Could you kindly help me to understand what `contractable` and `train_estimate` capabilities mean? I am refactoring ROCKERClassifier and I am not sure what they mean. @MatthewMiddlehurst

Reply:
`contractable` The user can set a time limit for the fit process. i.e. in cBOSS instead of searching a set number of parameter sets, keep searching parameter sets until time runs out.

`train_estimate` The classifier can generate a set of estimated probabilites for the training dataset. i.e. in cBOSS again, cross-validation is used to determine how useful an ensemble member is. We can reuse this if an estimate of performance is required on the training set. See also: out-of-bag estimates for forest classifiers.

Reply:
so for the descriptions, how about this:

contractable - `does the estimator support time limits on execution, aka contract time setting?`
train_estimate - `can the classifier produce out-of-bag predictions on the training set?`

Might be worth also adding an explicit mention to the method/argument if that's easy. What do you think, @MatthewMiddlehurst, on this as description strings?

Reply:
looks like contractable is not self-explanatory enough, if nobody is against it, I would like to rename it into `capability:set-max-fit-time` after TSC refactoring completion and add TODO comment about it

Reply:
For `contractable`, though, I think is better to emphasize that only train time can be limited, not predict, so `does the estimator support limiting max fit time?`
cbf10004

Reply:
I like `can the classifier produce out-of-bag predictions on the training set?` description for `train_estimate`
e94bec0a

Reply:
Worth noting that while out-of-bag predictions are the best example, they aren't the only method of generating train set predictions. Maybe replace out-of-bag with bespoke?

Contracting description is good.

Reply:
1. If I understood Matthew right, `train_estimate` description should be generalized to cover other methods to estimate classifier performance. Changed it to `can the classifier estimate its performance on the training set?`. If there are better suggestions, will gladly change it. 4cbd4398
2. It looks like `contract time setting` is a synonym term for `contractable`. If so, better add it to description. `contract time setting, i.e. does the estimator support limiting max fit time?` e84da651